### PR TITLE
Remove chemistry dep warning

### PR DESCRIPTION
**Pull request by kratman**
# Description

Remove the old deprecation warning for the chem argument. It was from March 2023, so over a year old

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Review by agriyakhetarpal — APPROVED** [review]

**Review by agriyakhetarpal — COMMENTED** [review]
I know that this is quite old and probably isn't in use anywhere either, but could you please add a CHANGELOG entry, too?

**Comment by kratman** [comment]
> I know that this is quite old and probably isn't in use anywhere either, but could you please add a CHANGELOG entry, too?

Yeah I was planning on it, I just needed to create the PR so I had the correct number

**Comment by agriyakhetarpal** [comment]
> > I know that this is quite old and probably isn't in use anywhere either, but could you please add a CHANGELOG entry, too?
> 
> Yeah I was planning on it, I just needed to create the PR so I had the correct number

You might like https://ichard26.github.io/next-pr-number/, I use it quite frequently for this purpose

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,8 @@
 
 - Added sensitivity calculation support for `pybamm.Simulation` and `pybamm.Experiment` ([#4415](https://github.com/pybamm-team/PyBaMM/pull/4415))
 - Added OpenMP parallelization to IDAKLU solver for lists of input parameters ([#4449](https://github.com/pybamm-team/PyBaMM/pull/4449))
-- Added phase-dependent particle options to LAM #4369
+- Added phase-dependent particle options to LAM
+  ([#4369](https://github.com/pybamm-team/PyBaMM/pull/4369))
```
Comment (arjxn-py) [inline CHANGELOG.md:8]:
```suggestion
- Added phase-dependent particle options to LAM ([#4369](https://github.com/pybamm-team/PyBaMM/pull/4369))
```

Comment (kratman) [inline CHANGELOG.md:8]: Fixed this one while I was updating my own log entry

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +19,8 @@
 
 ## Breaking changes
 
+- Removed the deprecation warning for the chemistry argument in
+  ParameterValues ([#4466](https://github.com/pybamm-team/PyBaMM/pull/4466))
```
Comment (arjxn-py) [inline CHANGELOG.md:23]:
```suggestion
- Removed the deprecation warning for the chemistry argument in ParameterValues ([#4466](https://github.com/pybamm-team/PyBaMM/pull/4466))
```

File: CHANGELOG.md (hunk 1 above)
Comment (kratman) [reply CHANGELOG.md:8]: Both of those render why does this matter?

Comment (arjxn-py) [reply CHANGELOG.md:8]: Should it not be in a single line? 😕

File: CHANGELOG.md (hunk 2 above)
Comment (kratman) [reply CHANGELOG.md:23]: Same as above

File: CHANGELOG.md (hunk 1 above)
Comment (agriyakhetarpal) [reply CHANGELOG.md:8]: I think it would split up into a new line on https://pybamm.org/changelog/

Comment (kratman) [reply CHANGELOG.md:8]: The wrapping was from my IDE, it does not make a difference in markdown as long as the link is not split between lines

Comment (kratman) [reply CHANGELOG.md:8]: It is already a bulleted list in that page, if that does not work then there is something wrong with the library used for it

Comment (agriyakhetarpal) [reply CHANGELOG.md:8]: Now that I notice, that page is outdated and does not show any results since 24.1 was last released

Comment (arjxn-py) [reply CHANGELOG.md:8]: I see, it's fine in that case.

Comment (agriyakhetarpal) [reply CHANGELOG.md:8]: In other words, the Goldmark renderer for Hugo has an option to wrap text, will check if it's enabled

**Review by arjxn-py — APPROVED** [review]
Thanks @kratman

**Pull request merged**